### PR TITLE
ExchangeContext remove unused flags

### PR DESCRIPTION
**Pull request by kghost**
#### Problem
Some legacy code is not being used at all

#### Change overview
* Rename `HasPiggybackAckPending` => `HasAckCounter`
* Clear up unused code

#### Testing
Passed unit-tests

### CODE REVIEW

**Review by andy31415 — APPROVED** [review]

File: src/messaging/ReliableMessageContext.h [hunk 1]
```diff
@@ -159,29 +138,24 @@ class ReliableMessageContext
...
-        kFlagAckPending = (1u << 5),
+        kFlagAckPending = (1u << 4),
 
-        /// When set, signifies that there has once been an acknowledgment
-        /// pending to be sent back.  In that case,
-        /// mPendingPeerAckMessageCounter is a valid message counter value for
-        /// some message we have needed to acknowledge in the past.
-        kFlagAckMessageCounterIsValid = (1u << 6),
+        /// When set, signifies that the message counter is valid.
+        /// The counter is valid once received a message with a request ack flag. It never flips to false.
```
Comment (bzbarsky-apple) [inline src/messaging/ReliableMessageContext.h:148]:
```suggestion
        /// When set, signifies that mPendingPeerAckMessageCounter is valid.
        /// The counter is valid once we receive a message which requests an ack.
        /// Once mPendingPeerAckMessageCounter is valid, it never stops being valid.
```

File: src/messaging/ReliableMessageContext.h [hunk 2]
```diff
@@ -67,11 +67,10 @@ class ReliableMessageContext
...
 
     /**
-     * Check whether we have an ack to piggyback on the message we are sending.
-     * If true, TakePendingPeerAckMessageCounter will return a valid value that
-     * should be included as an ack in the message.
+     * Check whether we have a valid message counter. The counter is valid once received a message with a request ack flag. It never
+     * flips to false.
      */
-    bool HasPiggybackAckPending() const;
+    bool HasAckCounter() const;
```
Comment (bzbarsky-apple) [inline src/messaging/ReliableMessageContext.h:73]: Why this change?  The old name accurately described the semantics of the function: if it returns true, there is a piggyback ack to be sent whenever we are sending a message.

**Comment by bzbarsky-apple** [comment]
@kghost are you still planning to get back to this?

**Review by bzbarsky-apple — APPROVED** [review]
